### PR TITLE
CODEOWNERS: add ownership for SIG Hubble API team

### PR DESCRIPTION
**Pull request by rolinh**
The new @cilium/sig-hubble-api team is responsible to review all Hubble API related changes. The Hubble API covers gRPC and metrics endpoints. The team ensures that API changes are backward compatible or that a new API version is created for backward incompatible changes.

### CODE REVIEW

File: CODEOWNERS
Comment (kaworu) [inline CODEOWNERS]: typo
```suggestion
#   Ensure the backwards-compatibility of Cilium REST and gRPC APIs, excluding
```

File: CODEOWNERS
Comment (aanm) [inline CODEOWNERS]: Why? :thinking:

File: CODEOWNERS
Comment (kaworu) [inline CODEOWNERS]: nit: since it's a CODEOWNER file
```suggestion
#   Hubble which is owned by @cilium/sig-hubble-api.
```

**Review by kaworu — CHANGES_REQUESTED** [review]

File: CODEOWNERS
Comment (kaworu) [reply CODEOWNERS]: @aanm I would guess because of the protoc stuff from the builder image.

Comment (rolinh) [reply CODEOWNERS]: We used to have a docker image (`images/hubble-proto`) up to Cilium v1.11 that contains the protobuf compiler as well as the protobuf plugins that we use. However, as of commit 2960b4e6fd6338b93a5865019eaf707915b88e96, `images/hubble-proto` was merged into `images/builder`. So now the Cilium builder image, which contains the protobuf compiler and the required plugins, [is used](https://github.com/cilium/cilium/blob/c84c03fa35df8a073759cb67b08b3eb10d791c14/api/v1/Makefile#L13) to generate the Go code from the proto definition.

Comment (aanm) [reply CODEOWNERS]: ok @rolinh but it kind of doesn't make sense to wait on two reviewers from sig-hubble-api to see for changes done in images/builder that doesn't modify the protobuf compiler.

Comment (rolinh) [reply CODEOWNERS]: Fwiw, I'm all for a separate proto builder image. Updating the cilium-builder image is [not a pleasant process](https://docs.cilium.io/en/v1.12/contributing/development/images/#update-cilim-builder-runtime-images) especially when another PR updates it at the same time and rebasing is required (and the whole triggering deployment has to start over again, etc). I am actually not convinced that merging `hubble-proto` with `builder` actually reduced maintenance overhead (it definitely didn't from my perspective as someone who has been updating the proto compiler and dependencies) and I'd be happy to re-create a hubble proto image.
cc @gandro

Comment (gandro) [reply CODEOWNERS]: Creating the hubble proto image is easy, maintaining is the hard part. The last one was unmaintained and outdated for almost a year, with no GitHub action available to update it.

There is a lot of infrastructure around each image (dependabot updates, go updates, maintaining the gh action workflow to push it). If you are willing to maintain all that, sure, please go a head, but I'm not sure what problem would be solved by a separate image. After all, we expect Cilium developers to have the builder image on hand.

As a compromise, I think it might be sufficient to just add `/images/builder/install-protoc.sh` and `/images/builder/install-protoplugins.sh` to the sig-hubble-api CODEOWNER here.

Comment (rolinh) [reply CODEOWNERS]: > Creating the hubble proto image is easy, maintaining is the hard part. The last one was unmaintained and outdated for almost a year, with no GitHub action available to update it.

It's no different with the builder as the base image, the proto deps are not automatically updated and afaik, I'm the only one having taken care of updating them "recently".

> I'm not sure what problem would be solved by a separate image

Less pain to update the image with proto updates and thus more incentive to take care of that.

> As a compromise, I think it might be sufficient to just add /images/builder/install-protoc.sh and /images/builder/install-protoplugins.sh to the sig-hubble-api CODEOWNER here.

Let's do that for now then.

Comment (gandro) [reply CODEOWNERS]: > Less pain to update the image with proto updates and thus more incentive to take care of that.

That's a valid argument. Again, I'm not opposed to unmerge, but if we unmerge, then let's do it properly this time (with documentation and a GH action to do it).

The last `hubble-proto` image didn't even have a Github Action to update it, leading us in a state where updates to had to happen manually from a developers laptop, pushing to DockerHub, with unclear ownership which developers even had the necessary access to do it. While the current process is painful, at least it's documented and can be done by any Cilium committer.

**Review by aanm — APPROVED** [review]

**Review by gandro — APPROVED** [review]

File: CODEOWNERS
Comment (rolinh) [reply CODEOWNERS]: > I'm not opposed to unmerge, but if we unmerge, then let's do it properly this time (with documentation and a GH action to do it).

Yes, definitely.

**Review by kaworu — APPROVED** [review]

**Comment by rolinh** [comment]
CODEOWNERS checks is :heavy_check_mark: and all required approvals are in, marking as ready to merge.

**Pull request merged**
